src/Jackalope/Version/VersionHandler.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+namespace Jackalope\Version;
+
+use Jackalope\Session;
+use Jackalope\Transport\AddNodeOperation;
+use Jackalope\Transport\WritingInterface;
+use PHPCR\InvalidItemStateException;
+use PHPCR\NodeInterface;
+use PHPCR\PropertyType;
+use PHPCR\UnsupportedRepositoryOperationException;
+use PHPCR\Util\UUIDHelper;
+use PHPCR\Version\VersionException;
+
+class VersionHandler
+{
+    const MIX_VERSIONABLE = 'mix:versionable';
+    const MIX_SIMPLE_VERSIONABLE = 'mix:simpleVersionable';
+
+    /**
+     * @var WritingInterface
+     */
+    private $objectManager;
+
+    /**
+     * @var Session
+     */
+    private $session;
+
+    public function __construct(Session $session)
+    {
+        $this->objectManager = $session->getObjectManager();
+        $this->session = $session;
+    }
+
+    public function addVersionProperties(NodeInterface $node)
+    {
+        $additionalOperations = array();
+
+        if ($node->hasProperty('jcr:isCheckedOut')) {
+            return $additionalOperations;
```

Comment:
lets move that before initialization of additionalOperations and `return array();` instead and comment "Node is already versioned, nothing to do".